File: SDKs/amazonaws.com/forecast/2018-06-26/python/src/sdk/models/shared/categoricalparameterrange.py
import dataclasses
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class CategoricalParameterRange:
    r"""CategoricalParameterRange
    Specifies a categorical hyperparameter and it's range of tunable values. This object is part of the <a>ParameterRanges</a> object.
    """
    
    name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('Name') }})
    values: list[str] = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('Values') }})
    

File: SDKs/amazonaws.com/forecast/2018-06-26/java/lib/src/main/java/openapisdk/models/shared/CategoricalParameterRange.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * CategoricalParameterRange
 * Specifies a categorical hyperparameter and it's range of tunable values. This object is part of the <a>ParameterRanges</a> object.
**/
public class CategoricalParameterRange {
    @JsonProperty("Name")
    public String name;
    public CategoricalParameterRange withName(String name) {
        this.name = name;
        return this;
    }
    @JsonProperty("Values")
    public String[] values;
    public CategoricalParameterRange withValues(String[] values) {
        this.values = values;
        return this;
    }
}